### ISSUE TITLE
Crash when selecting non-eligible vehicle part for install

### ISSUE THREAD

**Issue by amystewart256** [CleverRaven/Cataclysm-DDA#24372]
Game version: 0.C-29302-g14fb40a (tiles)

Operating system: Windows 10

Tiles or curses: Tiles

Mods active: More Survival Tools, Mutant NPCs, Folding Parts Pack, Boats, Vehicle Additions

### Expected behavior

Game doesn't crash

### Actual behavior

Game crashes

### Steps to reproduce the behavior

Examine a vehicle. Push i to open installation menu. Move the cursor down to select a grayed-out part (not eligible for installation). Game crashes every time.

![image](https://user-images.githubusercontent.com/7119741/42954298-5a47ddac-8b4a-11e8-9f0f-854aaefb0d77.png)

**Comment by mlangsdorf** [comment]
The image actually shows you installing a valid part - the requirements, tools, additional requirements, and description are all green.

I can't reproduce on current top of tree on Linux. I debugged in a character with an atomic lamp and a toolbox, then installed the atomic lamp on a vehicle. I then tried installing random other things on the vehicle and couldn't. Game didn't crash.

**Comment by mlangsdorf** [comment]
Dupe of #24373 24373

**Comment by amystewart256** [comment]
Yeah. The crash occurs during the act of scrolling down from a valid to invalid part, and seems to happen before the render, which is why a valid part is highlighted.

**Comment by amystewart256** [comment]
Addendum: in trying to repro this issue again, game has started crashing when I hit 'i' to open the installation menu (before the list even renders)

![image](https://user-images.githubusercontent.com/7119741/42956452-fcf61a34-8b4d-11e8-88c0-8472d1aed55f.png)

**Comment by mlangsdorf** [comment]
Not sure what's up with that.
but the initial bug is the lack of a description in storage_battery and some other items, and the lack of check before trying to print the description to make sure it exists.
I'm going to add the check and see if that helps, which I expect it will.

**Comment by mlangsdorf** [comment]
Thanks for the bug report. I think the fix I just pushed will solve the issue but please tell me if it doesn't.
